agent/agent.go

```diff
@@ -130,6 +128,22 @@ func (a *Agent) loop(ctx context.Context) error {
 	}
 }
 
+func makeTaskContext(nextTask *apimodels.NextTaskResponse, tc *taskContext) taskContext {
+	setupTeardownGroup := false
+	if nextTask.TaskGroup == "" || nextTask.TaskGroup != tc.taskGroup {
+		setupTeardownGroup = true
```

Comment:
should we also change/reset the task directory name in this case, or set it to use the task group-name rather than the task name when possible. 

empty taskGroup is just backwards compatibility for the transition, right?